custom/inddex/reports/nutrient_stats.py

```diff
@@ -26,24 +28,73 @@ def fields(self):
         ]
 
     @property
-    def report_config(self):
-        report_config = {}  # TODO port to FoodData.from_request
-        request_slugs = [
-            'gender',
-            'age_range',
-            'pregnant',
-            'breastfeeding',
-            'urban_rural',
-            'supplements',
-            'recall_status',
-        ]
-        report_config.update({slug: self.request.GET.get(slug, '')
-                              for slug in request_slugs})
-        return report_config
+    def data_providers(self):
+        food_data = FoodData.from_request(self.domain, self.request)
+        return [NutrientStatsData(food_data)]
+
+
+class NutrientStatsData:
+    title = 'Nutrient Intake Summary Stats'
+    slug = 'nutr_intake_summary_stats'
+
+    def __init__(self, food_data):
+        self._food_data = food_data
+        self._nutrient_names = self._food_data.fixtures.nutrient_names
 
     @property
-    @memoized
-    def data_providers(self):
+    def headers(self):
         return [
-            SummaryStatsNutrientDataProvider(config=self.report_config)
+            'nutrient', 'mean', 'median', 'std_dev', 'percentile_05',
+            'percentile_25', 'percentile_50', 'percentile_75', 'percentile_95'
         ]
+
+    @property
+    def rows(self):
+        for nutrient, amts in self._get_daily_totals():
+            yield format_row([
+                nutrient,
+                mean(amts),
+                median(amts),
+                stdev(amts),
+                percentile(amts, .05),
+                percentile(amts, .25),
+                percentile(amts, .5),
+                percentile(amts, .75),
+                percentile(amts, .95),
+            ])
+
+    def _get_daily_totals(self):
+        """For each nutrient, returns a list of daily totals for individual respondents"""
+        totals = defaultdict(lambda: defaultdict(int))
+        for row in self._food_data.rows:
+            for nutrient in self._nutrient_names:
+                # Keep a total amt for this respondent on this day
+                day_key = (row.unique_respondent_id, row.recalled_date)
+                totals[nutrient][day_key] += row.get_nutrient_amt(nutrient) or 0
+
+        for nutrient in self._nutrient_names:
+            yield nutrient, list(sorted(totals[nutrient].values()))
+
+
+def percentile(items, q):
```

Comment:
Numpy is no longer a dependency in HQ, and this seemed too small to warrant reintroducing it.